### ISSUE TITLE
Swiping Pages

### ISSUE THREAD

**Issue by DManstrator** [mobileappdevhm19/HINT-Reader#28]
I try to refactor my old code to make swiping possible without the usage of Dismissables.

**Comment by DManstrator** [comment]
So far I was able to detect Swipes but I didn't find out how to "goto" to the next or previous page. However we could remove the Font Size Buttons, when checking for Swipes I found a way to detect Zooms and I already implemented this step-wise zoom we had with the buttons, now with Gestures instead (see 2dac19f) .

To be honest, this issue is not really about swiping Pages only anymore, more like recognizing Gestures at all so I would rename this issue and try to cover both use cases here. Is that okay for you?

I also didn't find out how to test gestures yet. A quick Google Search gave me [this](https://github.com/flutter/flutter/blob/master/packages/flutter/test/widgets/gesture_detector_test.dart) and [that](https://github.com/flutter/flutter/blob/master/packages/flutter/test/gestures/gesture_binding_test.dart), maybe I can use it. See you in class then I guess. :D

**Comment by DManstrator** [comment]
So, current state:
I found a way to "goto" another Page by using a ScrollController and also implemented that. For that I needed to adjust how we build the ListView but in the end, it's the same. While doing that I refactored the existing functions a bit, guess it doesn't matter tho.

Something positive: Using the ScrollController, we can use animations and we even define to which position we want to go (so it would even be possible to skip pages with that). That works because of using offsets and adding or subtracting a value.

The negative part: I tried to figure out how we can access the height of a picture, without success. Even worse: I just looked at the pictures height myself and hardcoded that value and that value was way too big (1 Swipe skipped like 2 pages) so it seems the height also depends on the device size..

I saw you found out a way how to determine the right size of the device, maybe we can use that to get the value with should be added / subtracted.

Also I didn't write any test for it yet, need to do that over the weekend or when the feature is fully working as expected.

TL; DR:
- Zooming perfectly works
- Swiping pseudo-works
- No tests yet

See you in class. :D

**Comment by wildeKaiserin** [comment]
To your negative part: I think it would be enough to call the method getNextPage() if the backend is ready. So you won't have the problem getting the height of a picture

**Comment by DManstrator** [comment]
This is very true. I forgot that the display of pages will be differently in the future so we may not even need the ScrollController anymore, just a different implementation of the functions the GestureDetector triggers.

So I guess for now we can leave it as how it is and I try to write Test Cases for it.

**Comment by Kjani01** [comment]
Thus there couldn't be any progress for some time in this issue, we will take another look at it duing the next sprint.

**Comment by maxiking203** [comment]
@DManstrator because you ask me on last friday about testing. How is it going, do you need help?

**Comment by DManstrator** [comment]
@maxiking203 I don't think so, I just asked on Friday because I was looking at your tests and was overwhelmed by all the stuff. I'll try to do what you recommended, hopefully I get some results until we meet on Friday.

I had no time yet (Weekend to Tuesday for Cryptology, yesterday and today Networking 2 and also had to help a friend today) but I took off tomorrow so I'm sure I can do something at least.

Thanks for asking. :)

**Comment by wildeKaiserin** [comment]
I think, right now, the functionality of swiping pages in the backend is missing. So since we get no new backend, this issue is on ice for now.
